packages/hardhat-core/test/internal/multi-process-mutex.ts

```diff
@@ -0,0 +1,86 @@
+import { expect } from "chai";
+import fs from "node:fs";
+import path from "node:path";
+import os from "node:os";
+import { MultiProcessMutex } from "../../src/internal/util/multi-process-mutex";
+
+describe("multi-process-mutex", () => {
+  const mutexName = "test-mutex";
+
+  it("should execute all the function in a sequential order, not in parallel", async () => {
+    // Since all the functions cannot be executed in parallel because of the mutex,
+    // the total execution time should be bigger than the sum of the execution times of each function.
+    const mutex = new MultiProcessMutex(mutexName); // Use default max mutex lifespan
+    const start = performance.now();
+
+    const ms = [500, 700, 1000];
+    await Promise.all([
+      mutex.use(async () => {
+        await waitMs(ms[0]);
+      }),
+      mutex.use(async () => {
+        await waitMs(ms[1]);
+      }),
+
+      mutex.use(async () => {
+        await waitMs(ms[2]);
+      }),
+    ]);
+
+    const end = performance.now();
+    const duration = end - start;
+
+    expect(duration).to.be.greaterThan(ms[0] + ms[1] + ms[2]);
+  });
+
+  it("should overwrite an old mutex file", async () => {
```

Comment:
This test is checking an implementation detail, and the important part of this (that an attempt to acquire the mutex won't have to wait until the function finished) is already covered by the next test. I think we should remove it.